lib/elixir/lib/code.ex

```diff
@@ -1349,31 +1349,65 @@ defmodule Code do
 
   def fetch_docs(module) when is_atom(module) do
     case :code.get_object_code(module) do
-      {_module, bin, _beam_path} -> do_fetch_docs(bin)
-      :error -> {:error, :module_not_found}
+      {_module, bin, beam_path} ->
+        case fetch_docs_from_beam(bin) do
+          {:error, :chunk_not_found} ->
+            dir = beam_path |> :filename.dirname() |> :filename.dirname()
+            path = :filename.join([dir, 'doc', 'chunks', '#{module}.chunk'])
+            fetch_docs_from_chunk(path)
+
+          other ->
+            other
+        end
+
+      :error ->
+        case :code.which(module) do
+          :preloaded ->
+            path = :filename.join([:code.lib_dir(:erts), 'doc', 'chunks', '#{module}.chunk'])
```

Comment:
It is possible to use the Path functions and binaries instead? The same for above.